presto-main/src/main/java/com/facebook/presto/operator/scalar/MapToMapCast.java

```diff
@@ -239,7 +244,11 @@ public static Block mapCast(
             }
         }
         else {
-            TypedSet typedSet = new TypedSet(toKeyType, fromMap.getPositionCount() / 2, "map-to-map cast");
+            boolean skipDuplicateKeyCheck = canSkipDuplicateKeysCheck(fromKeyType, toKeyType);
+            TypedSet typedSet = null;
+            if (!skipDuplicateKeyCheck) {
```

Comment:
instead of calling the function skip and using it always negated, perhaps invert the logic and call it requiresDuplicateKeyCheck

Reply:
Done